libs/core/src/components/pds-popover/pds-popover.tsx

```diff
@@ -0,0 +1,220 @@
+import { Component, Element, Event, EventEmitter, Host, Listen, h, Method, Prop, State , Build} from '@stencil/core';
+import { TooltipPlacementType } from '@utils/types';
+
+@Component({
+  tag: 'pds-popover',
+  styleUrl: 'pds-popover.scss',
+  shadow: true,
+})
+export class PdsPopover {
+  /**
+   * Reference to the Host element
+   */
+  @Element() el: HTMLPdsPopoverElement;
+
+  /**
+   * Determines when the popover is active
+   * @defaultValue false
+   */
+  @State() active = false;
+
+  /**
+   * Determines the action that triggers the popover
+   * @defaultValue "show"
+   */
+  @Prop() popoverTargetAction: 'show' | 'hide' = 'show';
+
+  /**
+   * Determines the type of popover. Auto popovers can be "light dismissed" by clicking outside of the popover.
+   * Manual popovers require the consumer to handle the visibility of the popover.
+   */
+  @Prop() popoverType: 'auto' | 'manual' = 'auto';
+
+  /**
+   * A unique identifier used for the underlying component `id` attribute.
+   */
+  @Prop() componentId: string;
+
+  /**
+   * Text that appears on the trigger element
+   */
+  @Prop() text: string;
+
+  /**
+   * Sets the maximum width of the popover content
+   * @defaultValue "352px"
+   */
+  @Prop() maxWidth?: string = '352px';
```

Comment:
Should we make this a number to assist with the validation? We can append the `unit type`